ingestion/src/metadata/ingestion/source/database/hive/metadata.py

```diff
@@ -182,9 +185,40 @@ def get_table_comment(  # pylint: disable=unused-argument
     return {"text": None}
 
 
+def get_impala_columns(self, connection, table_name, schema=None, **kwargs):
+    """
+    Extracted from the Impala Dialect. We'll tune the implementation.
+
+    By default, this gives us the column name as `table.column`. We just
+    want to get `column`.
+    """
+    # pylint: disable=unused-argument
+    full_table_name = f"{schema}.{table_name}" if schema is not None else table_name
+    query = f"SELECT * FROM {full_table_name} LIMIT 0"
+    cursor = connection.execute(query)
+    schema = cursor.cursor.description
+    # We need to fetch the empty results otherwise these queries remain in
+    # flight
+    cursor.fetchall()
+    column_info = []
+    for col in schema:
+        column_info.append(
+            {
+                "name": remove_table_from_column_name(table_name, col[0]),
+                "type": _impala_type_to_sqlalchemy_type[col[1]],
```

Comment:
I'm not 100% sure if this would always work because there are many types which are supported by hive but not by the impala for example `binary` in that case it might fail unless the query returns something else for a binary `datatype` 🤔

Reply:
thanks for the catch. Updated using Hive's type map